src/mds/MDCache.cc

```diff
@@ -5449,10 +5452,12 @@ bool MDCache::process_imported_caps()
 	rejoin_session_map.empty()) {
       C_MDC_RejoinSessionsOpened *finish = new C_MDC_RejoinSessionsOpened(this);
       version_t pv = mds->server->prepare_force_open_sessions(rejoin_client_map,
+							      rejoin_client_metadata_map,
 							      finish->session_map);
-      mds->mdlog->start_submit_entry(new ESessions(pv, rejoin_client_map), finish);
+      ESessions *le = new ESessions(pv, std::move(rejoin_client_map),
+				    std::move(rejoin_client_metadata_map));
```

Comment:
Based on my understanding from [1], it is necessary to folow this up with `rejoin_client_map.clear()` and `rejoin_client_metadata_map.clear()` even though it's probably redundant. The standard apparently says that the containers are in an unspecified but **valid** state. So, since we want them to be empty after creating `ESessions`, just clear them after the move.

[1] https://stackoverflow.com/questions/28205387/is-it-safe-to-reuse-a-std-container-after-stdmove